src/persist-client/src/internal/state_serde.json

```diff
@@ -0,0 +1,209 @@
+{
+  "applier_version": "1.2.3",
+  "shard_id": "s0496ebd7-5c07-f7cd-1499-e74ed5c9bb03",
+  "seqno": 1773488203606603407,
+  "walltime_ms": 12135341112053470409,
+  "hostname": "𞸴ᜲ&ã/𝋊{'`𐖩%<ֈⶻ2 ﹪¥9>𝔚𞸁𖨓=e&᳆ \\",
+  "last_gc_req": 18060593127164614686,
+  "rollups": {
+    "12442657285421843877": {
+      "key": "\"L𚿺.\\.𑇡/Ѩ:𞟰﷿ꡥ𛰸୍൚ບ'iâ𑊆",
+      "encoded_size_bytes": 15119688182297364901
+    }
+  },
+  "leased_readers": {
+    "r1c270897-1cf0-76d4-0326-aac136fce468": {
+      "seqno": 12010739974165883287,
+      "since": {
+        "elements": []
+      },
+      "last_heartbeat_timestamp_ms": 9750575345611729810,
+      "lease_duration_ms": 8672934464634687609,
+      "debug": {
+        "hostname": "ᝒJ𞅇𬒤*ভ𑆱￭ln᳨[i{E",
+        "purpose": "𐔞�Wè𑼌𝡋_¥07¥𑄎:6"
+      }
+    },
+    "r7c77309d-fcd3-e0e7-1ffc-e5aff428513e": {
+      "seqno": 16186941177729540512,
+      "since": {
+        "elements": [
+          4355422458483136946
+        ]
+      },
+      "last_heartbeat_timestamp_ms": 3465149119092072769,
+      "lease_duration_ms": 7032165947814541071,
+      "debug": {
+        "hostname": "\\/𞹙ၹù;🅼.Fആ ಳΆÈ{:*n\"fຄ\"ဨೣ'𞤚𑿀x𞄻c",
+        "purpose": "𝍅'🠈Ἒ~{𝈝'எA&{𐨅.p:୳(¥{4<"
+      }
+    }
+  },
+  "critical_readers": {
+    "c7eced73b-1e76-2209-e97c-55607ca2b4c7": {
+      "since": {
+        "elements": [
+          12751762127056839801
+        ]
+      },
+      "opaque": 17666895704826768005,
+      "opaque_codec": "#𛅕Ἔ¥-R*¥௱�Wl\\q",
+      "debug": {
+        "hostname": "Ѩྦྷ𝨂ꨛ%𞹙w*ð𑌰{?`ି:`]𐽸/𑅜𑆂",
+        "purpose": "ধ𑁴𐀽º`FJ{�᮳𐭓Ѩ%fե𑈂ᛮ`Ⱥ"
+      }
+    }
+  },
+  "writers": {
+    "w402f07bd-d3de-e3ee-e4d0-175dde5dada6": {
+      "last_heartbeat_timestamp_ms": 10861220890935538203,
+      "lease_duration_ms": 16929229737484265737,
+      "most_recent_write_token": "i2812c6e2-8f0c-634a-25b1-e7c86c356ec3",
+      "most_recent_write_upper": {
+        "elements": [
+          16112463376238818326
+        ]
+      },
+      "debug": {
+        "hostname": "𞹤𑲰\\2¥<\"ங꣫꧘ዂ-Ѩ﹨\\Uዉ𛲈𝪮𞸧𑊲𫑈%/",
+        "purpose": "4X1S:y/:=\"X{ꬑJ/𐼧^^?/ؙ$[🕴"
+      }
+    },
+    "wab3862c1-4d86-1944-af27-d84849ace4d2": {
+      "last_heartbeat_timestamp_ms": 12250979085318804114,
+      "lease_duration_ms": 5379386706247460431,
+      "most_recent_write_token": "i70f486e4-6267-d6a5-8418-5bc446ee97ba",
+      "most_recent_write_upper": {
+        "elements": [
+          3814135578248638592
+        ]
+      },
+      "debug": {
+        "hostname": "🕴̏",
+        "purpose": ":Z𑜴\"𞋿𖿰1'Ѩ"
+      }
+    }
+  },
+  "since": [
```

Comment:
question for reviewers: if we require a `T: timely::order::TotalOrder` bound, then we can pull all these frontiers up to be non-array values. so e.g. `"since": [ 16259358130896200968 ]` -> `"since": 16259358130896200968`. worth it? it certainly saves quite a bit of noise when the json is pretty printed, but dunno how much we should be optimizing for that

Reply:
I'm partial to the current output, I don't find the extra noise problematic

Reply:
I do slightly prefer the less-noisy output, and also think that it's very unlikely that we ever have a non-totally-ordered timestamp, especially one that's visible to SQL. But I don't feel strongly; I think the main audience for this is likely to be `jq` or something anyways where it's almost moot.

Reply:
funny enough, I was thinking of `jq` slash some future where this is available in a way we can process it in mz. in particular, feels like we'll have to handle the potential for there to be multiple timestamps in like any processing that wants to deal with a frontier, which could be annoying/repetitive

i'm personally pulled strongly in both directions. i have a clear preference for the simpler output, but i also hesitate to introduce any sort of TotalOrder bounds in persist (beyond the one in shard_source that would be great to get rid of)

so, i'm split and we've got 1 vote each way from the two of you. given that this is definitely something than can evolve over time, seems like we should just pick one and revisit. in which case, i think it makes sense to try things first as-is and see how annoying the extra array is in practice